scripts/behavior.js

```diff
@@ -371,7 +371,129 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
                 }).then(
                     callback(COMMAND_SUCCESS)
                 );
-                
+
+                break;
+
+            case "createrole":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let roleName = escapeArguments(arguments);
+
+                let specificsCreateRole = utils.getSpecifics(message.guild);
+
+                if(specificsCreateRole["registeredRoles"].some(role => role === roleName)) {
+                    replyToMessage(message, "Role already registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                specificsCreateRole["registeredRoles"].push(roleName);
+                utils.writeSpecifics(message.guild, specificsCreateRole);
+
+                message.guild.createRole({name: arguments, color: "LIGHT_GREY", mentionable: true, hoist: false}, "Created by Dostya as requested by" + message.author.username)
+                    .then(value => {
+                        replyToMessage(message, "Role created")
+                            .then(callback(COMMAND_SUCCESS));
+                    }, rejectReason => {
+                        let specificsCreateRoleFailed = utils.getSpecifics(message.guild);
+                        delete specificsCreateRoleFailed.registeredRoles[specificsCreateRoleFailed.registeredRoles.indexOf(roleName)];
+                        utils.writeSpecifics(message.guild, specificsCreateRoleFailed);
+                        replyToMessage(message, "Could not create role due to internal discord error. " + rejectReason).then(callback(COMMAND_FORBIDDEN));
+                    });
+
+                break;
+
+            case "deleterole":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let delRoleName = escapeArguments(arguments);
+
+                let specificsDeleteRole = utils.getSpecifics(message.guild);
+
+                if(! specificsDeleteRole["registeredRoles"].some(role => role === delRoleName)) {
+                    replyToMessage(message, "Role not registered.")

```

Comment:
"Unknown role. Are you sure the spelling is correct ?"

Reply:
Ok, true, this is a user repsonse, got too technical on that one.